core/templates/pages/teach-page/teach-page.component.ts

```diff
@@ -15,64 +15,153 @@
 import { SiteAnalyticsService } from 'services/site-analytics.service';
-
+import { WindowDimensionsService } from 'services/contextual/window-dimensions.service.ts';
+import { LoaderService } from 'services/loader.service.ts';
+import { UserService } from 'services/user.service';
 @Component({
   selector: 'teach-page',
   templateUrl: './teach-page.component.html',
   styleUrls: []
 })
 export class TeachPageComponent implements OnInit {
-  TAB_ID_TEACH: string = 'teach';
-  TAB_ID_PARTICIPATION: string = 'participation';
-  TEACH_FORM_URL: string = 'https://goo.gl/forms/0p3Axuw5tLjTfiri1';
-  ALLOWED_TABS: string[] = [this.TAB_ID_TEACH, this.TAB_ID_PARTICIPATION];
-  activeTabName: string = this.TAB_ID_TEACH;
-
+  isWindowNarrow: boolean = false;
+  classroomUrlFragment: string;
+  classroomUrl :string;
+  displayedTestimonialId: number;
+  testimonialCount: number;
+  testimonials = [];
+  userIsLoggedIn: boolean = null;
   constructor(
     private siteAnalyticsService: SiteAnalyticsService,
     private urlInterpolationService: UrlInterpolationService,
-    private windowRef: WindowRef
+    private windowDimensionService: WindowDimensionsService,
+    private windowRef: WindowRef,
+    private userService: UserService,
+    private loaderService: LoaderService
   ) {}
 
   ngOnInit(): void {
-    const hash = this.windowRef.nativeWindow.location.hash.slice(1);
-    if (this.ALLOWED_TABS.includes(hash)) {
-      this.activeTabName = hash;
-    }
-    this.windowRef.nativeWindow.onhashchange = () => {
-      const hashChange = this.windowRef.nativeWindow.location.hash.slice(1);
-      if (this.ALLOWED_TABS.includes(hashChange)) {
-        this.activeTabName = hashChange;
-      }
-    };
+    this.isWindowNarrow = this.windowDimensionService.isWindowNarrow();
+    this.displayedTestimonialId = 0;
+    this.testimonialCount = 4;
+    this.testimonials = this.getTestimonials();
+    this.classroomUrl = this.urlInterpolationService.interpolateUrl(
+      '/learn/<classroomUrlFragment>', {
+        classroomUrlFragment: splashConstants.DEFAULT_CLASSROOM_URL_FRAGMENT
+      });
+    this.loaderService.showLoadingScreen('Loading');
+    this.userService.getUserInfoAsync().then((userInfo) => {
+      this.userIsLoggedIn = userInfo.isLoggedIn();
+      this.loaderService.hideLoadingScreen();
+    });
   }
 
-  onTabClick(tabName: string): Window {
-    // Update URL hash.
-    this.windowRef.nativeWindow.location.hash = '#' + tabName;
-    this.activeTabName = tabName;
-    return this.windowRef.nativeWindow;
+  // The 2 functions below are to cycle between values:
+  // 0 to (testimonialCount - 1) for displayedTestimonialId.
+  incrementDisplayedTestimonialId(): void {
+  // This makes sure that incrementing from (testimonialCount - 1)
+  // returns 0 instead of testimonialCount,since we want the testimonials
+  // to cycle through.
+    this.displayedTestimonialId = (
+      this.displayedTestimonialId + 1) % this.testimonialCount;
```

Comment:
Similar code is already used in the splash page, should we perhaps extract the code into a custom component?

Reply:
It is only 5-6 lines code and is very simple to copy. So , maybe it wont be much beneficial to make a custom component.

Reply:
I mean the whole testimonials code is about 40 lines? Which might be worth extracting? It would be basically a component where you can put a list of testimonials and it would display it.

Reply:
Okay, sure, 'll make a separate component.
Can we proceed with this code for now? I will make a separate component in another PR and then change  the code in splash page, teach page  and the about page  (#11420 ) (It is yet to be merged) in the same. Should i add a "To do" about this above the testinomials code for now?

Reply:
Yes, I'm fine with that. Please add a TODO with an issue here as a comment.

Reply:
done.

Reply:
Please add the TODO also in the code `// TODO(#11657): Separate testimonitals into component.`

Reply:
I have already added in line no.67 .
![image](https://user-images.githubusercontent.com/68547101/104148918-39b49f00-53fa-11eb-887f-2c55448150c2.png)

Reply:
Yep, sorry.